tools/ci/tc/decision.py

```diff
@@ -0,0 +1,314 @@
+import argparse
+import json
+import logging
+import os
+import re
+import subprocess
+from collections import OrderedDict
+
+import taskcluster
+from six import iteritems, itervalues
+
+from . import taskgraph
+
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+logging.basicConfig()
+logger = logging.getLogger()
+
+
+def get_triggers(event):
+    # Set some variables that we use to get the commits on the current branch
+    ref_prefix = "refs/heads/"
+    pull_request = "pull_request" in event
+    branch = None
+    if not pull_request and "ref" in event:
+        branch = event["ref"]
+        if branch.startswith(ref_prefix):
+            branch = branch[len(ref_prefix):]
+
+    return pull_request, branch
+
+
+def fetch_event_data(queue):
+    try:
+        task_id = os.environ["TASK_ID"]
+    except KeyError:
+        logger.warning("Missing TASK_ID environment variable")
+        # For example under local testing
+        return None
+
+    task_data = queue.task(task_id)
+
+    event_data = task_data.get("extra", {}).get("github_event")
+    if event_data is not None:
+        return event_data
+
+
+def filter_triggers(event, all_tasks):
+    is_pr, branch = get_triggers(event)
+    triggered = {}
+    for name, task in iteritems(all_tasks):
+        if "trigger" in task:
+            if is_pr and "pull-request" in task["trigger"]:
+                triggered[name] = task
+            elif branch is not None and "branch" in task["trigger"]:
+                for trigger_branch in task["trigger"]["branch"]:
+                    if (trigger_branch == branch or
+                        trigger_branch.endswith("*") and branch.startswith(trigger_branch[:-1])):
+                        triggered[name] = task
+    logger.info("Triggers match tasks:\n * %s" % "\n * ".join(triggered.keys()))
+    return triggered
+
+
+def get_run_jobs(event):
+    from tools.ci import jobs
+    revish = "%s..%s" % (event["pull_request"]["base"]["sha"]
+                         if "pull_request" in event
+                         else event["before"],
+                         event["after"])
+    logger.info("Looking for changes in range %s" % revish)
+    paths = jobs.get_paths(revish=revish)
+    logger.info("Found changes in paths:%s" % "\n".join(paths))
+    path_jobs = jobs.get_jobs(paths)
+    all_jobs = path_jobs | get_extra_jobs(event)
+    logger.info("Including jobs:\n * %s" % "\n * ".join(all_jobs))
+    return all_jobs
+
+
+def get_extra_jobs(event):
+    body = None
+    jobs = set()
+    if "commits" in event and event["commits"]:
+        body = event["commits"][0]["message"]
+    elif "pull_request" in event:
+        body = event["pull_request"]["body"]
+
+    if not body:
+        return jobs
+
+    regexp = re.compile(r"\s*tc-jobs:(.*)$")
```

Comment:
The PR description talks about this a potential future improvement, but it already existed and is kept working here :)

Should it be documented somewhere? I only know it exists because I've seen it in the code, but I've never seen it used, even by you :)

Reply:
Works for me, but can you update the commit message around "could invent a tc-try:<jobs> syntax"?